stone/backends/python_types.py

```diff
@@ -224,14 +230,92 @@ def _class_declaration_for_type(self, ns, data_type):
         if data_type.parent_type:
             extends = class_name_for_data_type(data_type.parent_type, ns)
         else:
-            if is_union_type(data_type):
+            if is_struct_type(data_type):
                 # Use a handwritten base class
+                extends = 'bb.Struct'
+            elif is_union_type(data_type):
                 extends = 'bb.Union'
             else:
                 extends = 'object'
         return 'class {}({}):'.format(
             class_name_for_data_type(data_type), extends)
 
+    #
+    # Annotation types
+    #
+
+    def _generate_annotation_type_class(self, ns, annotation_type):
+        # type: (ApiNamespace, AnnotationType) -> None
+        """Defines a Python class that represents an annotation type in Stone."""
+        self.emit('class {}(bb.AnnotationType):'.format(
+            class_name_for_annotation_type(annotation_type, ns)))
+        with self.indent():
+            if annotation_type.has_documented_type_or_params():
+                self.emit('"""')
+                if annotation_type.doc:
+                    self.emit_wrapped_text(
+                        self.process_doc(annotation_type.doc, self._docf))
+                    if annotation_type.has_documented_params():
+                        self.emit()
+                for param in annotation_type.params:
+                    if not param.doc:
+                        continue
+                    self.emit_wrapped_text(':ivar {}: {}'.format(
+                        fmt_var(param.name),
```

Comment:
A few `fmt_var` are not called with `check_reserved` set to `True`. This can lead to inconsistency when the name is a Python keyword.